string trans_inputs
---
string current_inputs_outputs